projects/RabbitMQ.Client/CreateChannelOptions.cs

```diff
@@ -16,15 +50,10 @@ public sealed class CreateChannelOptions
         public bool PublisherConfirmationTrackingEnabled { get; set; } = false;
 
         /// <summary>
-        /// If publisher confirmation tracking is enabled, this represents the number of allowed
-        /// outstanding publisher confirmations before publishing is blocked.
-        ///
-        /// Defaults to <c>128</c>
-        ///
-        /// Set to <c>null</c>, to allow an unlimited number of outstanding confirmations.
-        ///
+        /// If the publisher confirmation tracking is enabled, this represents the rate limiter used to
+        /// throttle additional attempts to publish once the threshold is reached.
         /// </summary>
-        public ushort? MaxOutstandingPublisherConfirmations { get; set; } = 128;
+        public RateLimiter? OutstandingPublisherConfirmationsRateLimiter { get; set; } = new ThrottlingRateLimiter(128);
```

Comment:
Probably all this could be further improved by setting the default implementation when the tracking is enabled and when the tracking is disabled add a No-Op implementation. Then the code might become super straightforward in the publisher confirm implementation